sql/expression_resolver.go

```diff
@@ -70,33 +70,39 @@ type gitLabModule struct {
 }
 
 type resolvedTrainClause struct {
-	IsPreMadeModel                   bool
-	ModelName                        string
-	ModelConstructorParams           map[string]*attribute
-	BatchSize                        int
-	EvalBatchSize                    int
-	DropRemainder                    bool
-	EnableCache                      bool
-	CachePath                        string
-	Epoch                            int
-	Shard                            int
-	EnableShuffle                    bool
-	ShuffleBufferSize                int
-	MaxSteps                         int
-	GradsToWait                      int
-	TensorboardLogDir                string
-	CheckpointSteps                  int
-	CheckpointDir                    string
-	KeepCheckpointMax                int
-	EvalSteps                        int
-	EvalStartDelay                   int
-	EvalThrottle                     int
-	EvalCheckpointFilenameForInit    string
-	PredictCheckpointFilenameForInit string
-	FeatureColumns                   map[string][]featureColumn
-	ColumnSpecs                      map[string][]*columnSpec
-	EngineParams                     engineSpec
-	CustomModule                     *gitLabModule
+	IsPreMadeModel                bool
+	ModelName                     string
+	ModelConstructorParams        map[string]*attribute
+	BatchSize                     int
+	EvalBatchSize                 int
+	DropRemainder                 bool
+	EnableCache                   bool
+	CachePath                     string
+	Epoch                         int
+	Shard                         int
+	EnableShuffle                 bool
+	ShuffleBufferSize             int
+	MaxSteps                      int
+	GradsToWait                   int
+	TensorboardLogDir             string
+	CheckpointSteps               int
+	CheckpointDir                 string
+	KeepCheckpointMax             int
+	EvalSteps                     int
+	EvalStartDelay                int
+	EvalThrottle                  int
+	EvalCheckpointFilenameForInit string
```

Comment:
It looks this field renamed to `CheckpointFilenameForInit` in `resolvedPredictClause`.
Shall we do the same thing here?

Reply:
I'd like to keep `Eval` here for now since currently training and evaluation is coupled and it would be good to keep `Eval` so that we know this is only used for evaluation. `resolvedPredictClause` is only related to prediction so it's unnecessary to name it `PredictCheckpointFilenameForInit `.